src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2232,6 +2234,36 @@ QString TorrentImpl::createMagnetURI() const
     return QString::fromStdString(lt::make_magnet_uri(m_nativeHandle));
 }
 
+nonstd::expected<void, QString> TorrentImpl::exportToFile(const Path &path) const
+{
+    if (!hasMetadata())
+        return nonstd::make_unexpected(tr("Missing metadata"));
+
+    try
+    {
+#ifdef QBT_USES_LIBTORRENT2
+        const std::shared_ptr<lt::torrent_info> torrentInfo = m_nativeHandle.torrent_file_with_hashes();
```

Comment:
#16896
It can return `nullptr` in case when hashes aren't completely available.

Reply:
Fixed.